llama_stack/cli/stack/_build.py

```diff
@@ -71,6 +71,10 @@ def run_stack_build_command(args: argparse.Namespace) -> None:
     else:
         image_name = args.image_name
 
+    if not image_name:
```

Comment:
I don't think this is right. Later in the code path we have an exception where we throw if you have not specified an image name but sadly that exception is only thrown for ImageType.conda. Can we always throw this exception? I think making up an image name is confusing, isn't it?